File: code/backend/src/main/java/project/backend/dto/RecipeDto.java
package project.backend.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RecipeDto {

    private Long id;

    private String name;

    private String description;

    private int difficulty;

    private int preparationTime;

    private String mealType;

    private String recipeCategory;

    private String[] images;

    private String[] tags;

    private RecipeIngredientListDto ingredients;

    private RecipeStepsDto steps;

    private int averageRating;

    private int numberOfRatings;
}

File: code/frontend/src/app/dtos/recipe-dto.ts
import { RecipeIngredientListDto } from "./recipe-ingredient-list-dto";
import { RecipeStepsDto } from "./recipe-steps-dto";

export interface RecipeDto {
  id: number;
  name: string;
  description: string;
  difficulty: number;
  preparationTime: number;
  mealType: string;
  recipeCategory: string;
  images: string[];
  tags: string[];
  ingredients: RecipeIngredientListDto;
  steps: RecipeStepsDto;
  averageRating: number;
  numberOfRatings: number;
}
